tests/unit/autocomplete/test_completer.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from awscli.testutils import unittest, mock
+from awscli.autocomplete import completer, parser, model
+from awscli.clidriver import CLIDriver
+
+
+class TestAutoCompleter(unittest.TestCase):
+    def setUp(self):
+        self.parser = mock.Mock(spec=parser.CLIParser)
+        self.parsed_result = parser.ParsedResult()
+        self.parser.parse.return_value = self.parsed_result
+
+    def test_delegates_to_autocompleters(self):
+        mock_complete = mock.Mock(spec=completer.BaseCompleter)
+        mock_complete.complete.return_value = ['ec2', 'ecs']
+        auto_complete = completer.AutoCompleter(
+            self.parser, completers=[mock_complete])
+
+        results = auto_complete.autocomplete('aws e')
+        self.assertEqual(results, ['ec2', 'ecs'])
+        self.parser.parse.assert_called_with('aws e', None)
+        mock_complete.complete.assert_called_with(self.parsed_result)
+
+    def test_stops_processing_when_list_returned(self):
```

Comment:
Do we want a test when both the first and second completers are set to return a value, but the first is the only one called?